src/Hosting/TestHost/src/RequestLifetimeFeature.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    internal class RequestLifetimeFeature : IHttpRequestLifetimeFeature
+    {
+        private readonly CancellationTokenSource _abortableCancellationTokenSource = new CancellationTokenSource();
+        private CancellationTokenSource _linkedCancellationTokenSource;
+
+        public CancellationToken RequestAborted
```

Comment:
It's the caller's responsibility to link tokens, and they may choose to link them in other ways. You can replace _linkedCancellationTokenSource with a field for the current CancellationToken initialized to _abortableCancellationTokenSource.Token.

```suggestion
        public CancellationToken RequestAborted { get; set; } = _abortableCancellationTokenSource.Token;
```

Reply:
So, when the caller passes a `CancellationToken`, the `Abort()` method stops working. That's very weird to me 🤔

Reply:
Yes, that's the behavior of all of our servers (though Abort also does other things in most servers like close the socket). The caller setting CancellationToken is intentionally customizing the behavior, they can choose how the linking happens.